### PR TITLE
Improve login scripts

### PR DESCRIPTION
**Pull request by emmanuelbernard**
- Improve rhoas_login with clipboard copy and dryrun
- Safe login script
- Add doc on login-all.sh

### CODE REVIEW

File: login-all.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+# make sure your OpenShift token has not expired
+# open URL to login to OpenShift via oc as needed
+# logs to rhoas
+
+. kas-installer.env
```
Comment (MikeEdgar) [inline login-all.sh:6]:
```suggestion
DIR_NAME="$(dirname $0)"
. ${DIR_NAME}/kas-installer.env
```

File: login-all.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+oc get route -n mas-sso keycloak > /dev/null 2>&1
+status=$?
+if [ $status -eq 1 ]; then
+    oc login --server=https://api.${K8S_CLUSTER_DOMAIN}:6443
+    open https://oauth-openshift.apps.${K8S_CLUSTER_DOMAIN}/oauth/token/request
+    exit 1;
+else
+    ./rhoas_login.sh
```
Comment (MikeEdgar) [inline login-all.sh:14]:
```suggestion
    ${DIR_NAME}/rhoas_login.sh
```

File: rhoas_login.sh [hunk 3]
```diff
@@ -3,10 +3,63 @@
...
- --auth-url $(oc get route -n mas-sso keycloak -o json | jq -r '"https://"+.spec.host+"/auth/realms/rhoas"')  \
- --mas-auth-url $(oc get route -n mas-sso keycloak -o json | jq -r '"https://"+.spec.host+"/auth/realms/rhoas"')  \
- --api-gateway $(oc get route -n kas-fleet-manager-${USER} kas-fleet-manager -o json | jq -r '"https://"+.spec.host')
+login() {
+    echo "Login with ${RH_USERNAME}/${RH_USERNAME}"
+    which pbcopy &>/dev/null
+    if  [ $? -eq 0 ]; then
+        echo "Copying ${RH_USERNAME} in the clipboard"
+        echo "${RH_USERNAME}" | pbcopy
+    fi
```
Comment (MikeEdgar) [inline rhoas_login.sh:13]:
```suggestion
    OS=$(uname)

    if [ "$OS" = 'Darwin' ]; then
        # for MacOS
        COPYCMD=$(which pbcopy &>/dev/null)
    else
        # for Linux and Windows
        COPYCMD=$(which xclip &>/dev/null)
        if [ -n "${COPYCMD}" ] ; then
            COPYCMD="${COPYCMD} -selection clipboard"
        fi
    fi

    if [ -n "${COPYCMD}" ] ; then
        echo "Copying ${RH_USERNAME} in the clipboard"
        echo "${RH_USERNAME}" | ${COPYCMD}
    fi
```

**Review by MikeEdgar — CHANGES_REQUESTED** [review]
@emmanuelbernard thanks for opening this. I have a few suggestions to avoid the assumption on the working directory and support Linux for the username copy feature as well.

File: rhoas_login.sh (hunk 3 above)
Comment (emmanuelbernard) [reply rhoas_login.sh:13]: Thanks I thought pbcopy was available on Linux too, my bad.

**Review by MikeEdgar — APPROVED** [review]
lgtm, thanks!

**Pull request merged**